src/Microsoft.DotNet.Build.Tasks/UpdateProjectDependencies.cs

```diff
@@ -0,0 +1,140 @@
+﻿using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using NuGet.Versioning;
+using System;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class UpdateProjectDependencies : Task
+    {
+        [Required]
+        public ITaskItem ProjectJson { get; set; }
+
+        public string OldPrerelease { get; set; }
+
+        public string NewPrerelease { get; set; }
+
+        public string PackageId { get; set; }
+
+        public string OldVersion { get; set; }
+
+        public string NewVersion { get; set; }
+
+        public override bool Execute()
+        {
+            bool anyPrereleaseUpdateParams = new[] { OldPrerelease, NewPrerelease }.Any(s => !string.IsNullOrEmpty(s));
+            bool anyVersionUpdateParams = new[] { PackageId, OldVersion, NewVersion }.Any(s => !string.IsNullOrEmpty(s));
+
+            if (anyPrereleaseUpdateParams ^ anyVersionUpdateParams)
+            {
+                if (new[] { OldPrerelease, NewPrerelease }.All(s => !string.IsNullOrEmpty(s)))
+                {
+                    ReplacePrerelease();
+                }
+                else if (new[] { PackageId, OldVersion, NewVersion }.All(s => !string.IsNullOrEmpty(s)))
+                {
+                    ReplacePackageVersion();
+                }
+                else
+                {
+                    if (anyPrereleaseUpdateParams)
+                    {
+                        Log.LogError("Not all properties found: expected (OldPrerelease, NewPrerelease)");
+                    }
+                    else
+                    {
+                        Log.LogError("Not all properties found: expected (PackageId, OldVersion, NewVersion)");
+                    }
+                }
+            }
+            else
+            {
+                Log.LogError(
+                    "Expected properties from one of (OldPrerelease, NewPrerelease) or (PackageId, OldVersion, NewVersion) but found {0}.",
+                    (anyPrereleaseUpdateParams && anyVersionUpdateParams) ? "properties from both" : "none");
+            }
+            return !Log.HasLoggedErrors;
```

Comment:
@ericstj Is the above parameter checking at all reasonable? Straightforward `if`s works but gets kinda big with 3 parameters (vs. reasonable for 2).